### PR TITLE
feat(web): two-phase semantic generate — instant baseline + cost-gated streamed enrichment

### LINKED ISSUE #3236: feat(web): two-phase semantic generate — instant mechanical baseline + cost-gated streamed enrichment
The "just works" core of onboarding: when you generate a semantic layer for a group, the **mechanical baseline appears instantly and free**, and **enrichment is an explicit, cost-controlled, streamed** upgrade — never an accidental large LLM bill.

## What to build

Two phases in the generate flow:

**Phase 1 — mechanical baseline (no LLM, auto, free, instant).** Profiler → complete entity/catalog/glossary/metric YAML for every non-ignored table. Usable and queryable on its own.

**Phase 2 — enrichment (explicit, never auto-fires).** LLM round-trip (via the shared engine, #3233) that may run read-only DB queries to ground its output. Controls:

- **Enrich all** — single action, behind a **confirmation that shows the scale** (e.g. "Enrich 24 tables?") so spend is never a surprise.
- **Enrich selected** — a per-table selector.
- **Ignore** — exclude tables you don't need; the ignore list is **pre-seeded** from the profiler's `table_flags.possibly_abandoned` (legacy / junction / cache), so you confirm rather than hunt.
- Results **stream in per table**, upgrading each row in place.

## Key files

- `packages/api/src/api/routes/wizard.ts` (`/generate` → mechanical; new enrich endpoint(s), streamed/per-table)
- `packages/web/src/app/wizard/page.tsx` (two-phase UI, enrich controls, progress/stream)
- shared engine (#3233); `table_flags.possibly_abandoned` from `packages/api/src/lib/profiler.ts`
- `docs/design/semantic-onboarding.md` (§ D)

## Acceptance criteria

- [ ] Generate produces a complete mechanical baseline with **no** LLM call; it's immediately saveable and queryable.
- [ ] Enrichment never fires automatically; "Enrich all" requires an explicit confirm stating the table count (plus any available cost/scale signal).
- [ ] Per-table selector enriches only chosen tables; ignored tables are excluded and not enriched.
- [ ] Ignore list is pre-seeded from `possibly_abandoned`; the user can adjust it.
- [ ] Enrichment results stream in per table and upgrade rows in place; partial completion is safe (save after baseline, enrichment continues/upgrades).
- [ ] Tests: mechanical-only path (no LLM), enrich-all confirm gate, enrich-selected, ignore-prefill, streamed/partial results.

## Dependencies

- Blocked by #3233 (shared generation engine — enrichment must be API-callable)
- Blocked by #3234 (generation writes into the group namespace)

### PR DESCRIPTION
**Pull request by msywulak**
Closes #3236.

The "just works" core of onboarding: generating a semantic layer for a group now produces an **instant, free, mechanical baseline**, and enrichment is an **explicit, cost-controlled, streamed** upgrade — never an accidental large LLM bill.

## Phase 1 — mechanical baseline (unchanged)
The existing `/wizard/generate` (profiler → complete entity YAML, **no LLM**) and `/wizard/save` are the baseline. It's immediately saveable and queryable on its own.

## Phase 2 — enrichment (new)
A per-table **`POST /api/v1/wizard/enrich`** route + a two-phase Review UI:

- **Enrich all** — one action behind a confirm dialog that **states the table count** (`Enrich N tables?`). Never auto-fires.
- **Enrich selected** — per-row checkboxes; enriches only the chosen, non-ignored tables.
- **Ignore** — pre-seeded from the profiler's `table_flags.possibly_abandoned` (so you confirm rather than hunt); ignored tables are excluded from **both** enrichment and the final save. User-adjustable.
- **Streamed per table** — the frontend fires one request per table with bounded concurrency (`runWithConcurrency`) and upgrades each row in place the instant it settles. Each table is independent, so **partial completion is safe** (save the baseline anytime; enrichment upgrades rows as it lands).
- **Fails fast** — when no LLM provider is configured the route returns `503` (`getMissingModelConfig`), so a key-less self-host shows **one** actionable banner instead of N per-table auth errors, and the baseline stays saveable.

## Shared engine (no CLI ↔ web drift)
The enrichment engine gains an in-memory **`enrichEntityYaml`**, extracted from the file-based `enrichEntity` (#3233). The file-based path now delegates to it (CLI behaviour unchanged), so `atlas init --enrich` and the wizard share the exact same prompt + deep-merge rules. The in-memory variant **throws** on a provider/network error (so the route can map it per-table) and **soft-skips** an unparseable response (returns the baseline with `enriched: false`).

Builds on #3234 (group-namespace generation) and #3286 — enrichment merges into the YAML `/generate` already scoped to the group, and `/save` keeps the group scoping; no group-routing changes here.

## Acceptance criteria
- [x] Mechanical baseline with **no** LLM; immediately saveable + queryable.
- [x] Enrichment never auto-fires; "Enrich all" requires an explicit confirm stating the table count.
- [x] Per-table selector enriches only chosen tables; ignored tables excluded + not enriched.
- [x] Ignore list pre-seeded from `possibly_abandoned`; user-adjustable.
- [x] Results stream in per table, upgrade rows in place; partial completion safe.
- [x] Tests: mechanical-only path (existing `/generate` no-LLM), enrich-all confirm-gate count, enrich-selected, ignore-prefill, streamed/partial.

## Tests
- **Engine** (`enrich-engine.test.ts`): `enrichEntityYaml` merge / soft-skip / throw-on-provider-error / usage accounting.
- **Route** (`wizard.test.ts`): `/enrich` happy path, no-usable-output passthrough, `503` no-provider (asserts it never profiles or enriches), `404`, `500`, `403`.
- **Frontend** (`wizard-enrich.test.ts`): ignore pre-fill, enrich-all target count, enrich-selected guard, bounded concurrency + partial safety.

## Gates
Full `bun run test` + `bun run type` + `bun run lint` all green. OpenAPI spec (`apps/docs/openapi.json`) + api-reference MDX regenerated for the new route.

## Key files
- `packages/api/src/api/routes/wizard.ts` — new `/enrich` route
- `packages/api/src/lib/semantic/enrich/index.ts` — `enrichEntityYaml`
- `packages/api/src/lib/providers.ts` — `getMissingModelConfig`
- `packages/web/src/app/wizard/page.tsx` — two-phase Review UI
- `packages/web/src/app/wizard/wizard-enrich.ts` — pure helpers
- `packages/types/src/profiler.ts` — `WizardEnrichResult`
- `docs/design/semantic-onboarding.md` — § D "as implemented (#3236)"

---
<a href="https://app.blacksmith.sh/AtlasDevHQ/codesmith/atlas/pr/3289"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-dark-v2.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-with-codesmith-light-v2.svg">View with Codesmith</picture></a> <a href="https://backend.blacksmith.sh/track/enable-autofix?expires=1783367727&installation_id=135337507&pr_number=3289&repository=AtlasDevHQ%2Fatlas&return_to=https%3A%2F%2Fgithub.com%2FAtlasDevHQ%2Fatlas%2Fpull%2F3289&signature=620e52412e09ea7c386023437847a5592196fdebe6f0e31c46af9867640b9b09"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/autofix-with-codesmith-light.svg">Autofix with Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>/codesmith</code> with what you need. Autofix is disabled.</sup>

## Summary by CodeRabbit

* **New Features**
  * Per-table LLM enrichment for entity metadata (adds business descriptions, use cases, query patterns, virtual dimensions)
  * UI controls: "Enrich selected" and confirmed "Enrich all" with per-table status, ignore/restore per table, and bounded concurrent enrichment
  * Admin-gated enrichment requiring a configured LLM provider and triggered only via explicit UI action
  * New POST enrichment API returning enriched YAML and an enriched flag

* **Documentation**
  * Updated onboarding/design doc describing the two-phase generate→enrich workflow

* **Tests**
  * New test coverage for enrichment flows, edge cases, and provider preflight checks

### CODE REVIEW

**Comment by msywulak** [comment]
@codex review

File: packages/api/src/api/routes/wizard.ts [hunk 1]
```diff
@@ -676,6 +753,123 @@ wizard.openapi(generateRoute, async (c) => {
...
+    const { requestId } = yield* RequestContext;
+    const { user } = yield* AuthContext;
+
+    const { connectionId, tableName, yaml: baselineYaml } = c.req.valid("json");
+
+    // Fail fast (503) when no LLM provider is configured so the UI surfaces ONE
+    // actionable banner instead of every per-table enrich hitting the same
+    // provider-auth error. Mirrors getModel()'s env-based provider resolution.
+    const { provider, missing } = getMissingModelConfig();
+    if (missing.length > 0) {
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/api/routes/wizard.ts]: Fixed in e167a6e0. \`/wizard/enrich\` now resolves the model exactly like the agent loop (lib/agent.ts): \`runEnterprise(ModelRouter.getWorkspaceModelConfigRaw)\` → \`getModelFromWorkspaceConfig\` first, falling back to the env provider. So an admin-configured workspace BYOT provider is used even with no platform env key; the 503 now only fires when neither a workspace key nor a platform provider is configured. Added a route test for the BYOT-without-env-key path.

File: packages/api/src/api/routes/wizard.ts [hunk 2]
```diff
@@ -676,6 +753,123 @@ wizard.openapi(generateRoute, async (c) => {
...
+        // samples/distributions (semantic-onboarding § D: enrichment "receives
+        // the table profile AND read-only access to the DB" — the profiler IS
+        // that read-only access).
+        let result: ProfilingResult;
+        switch (dbType) {
+          case "postgres":
+            result = await profilePostgres(url, [tableName], undefined, schema, undefined, log);
+            break;
+          case "mysql":
+            result = await profileMySQL(url, [tableName], undefined, undefined, log);
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/api/routes/wizard.ts:832]: Fixed in e167a6e0. \`enrichEntityYaml\` now takes the datasource \`dbType\` and the prompt requests MySQL vs PostgreSQL syntax accordingly; the route threads the resolved \`dbType\` through. The file-based CLI path defaults to PostgreSQL (Postgres-oriented). Added engine tests asserting the dialect in the prompt.

File: packages/web/src/app/wizard/page.tsx [hunk 3]
```diff
@@ -638,9 +690,149 @@ function StepReview({
...
+        }
+        if (outcome.kind === "error") {
+          setEnrichStatus((prev) => ({ ...prev, [tableName]: "error" }));
+          setEnrichRowError((prev) => ({ ...prev, [tableName]: outcome.message }));
+          if (outcome.unavailable) {
+            setEnrichError({ message: outcome.message, requestId: outcome.requestId });
+          }
+          return;
+        }
+        setEnrichStatus((prev) => ({ ...prev, [tableName]: "enriched" }));
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/web/src/app/wizard/page.tsx:865]: Fixed in e167a6e0. When the endpoint returns \`enriched: false\`, the row now shows a muted "No changes" badge (with a tooltip) and keeps its mechanical baseline, instead of an "Enriched" badge.

File: packages/web/src/app/wizard/page.tsx [hunk 4]
```diff
@@ -638,9 +690,149 @@ function StepReview({
...
+        handleYamlChange(tableName, outcome.yaml);
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/web/src/app/wizard/page.tsx]: Fixed in e167a6e0. The YAML textarea is now locked while its row is enriching, and the settle path (\`applyEnrichedYaml\`) only writes the result if the row still matches the YAML we sent — a manual edit made mid-flight wins instead of being clobbered.

File: apps/docs/openapi.json [hunk 5]
```diff
@@ -14135,6 +14135,205 @@
...
+                },
+                "required": [
+                  "connectionId",
+                  "tableName",
+                  "yaml"
+                ]
+              }
+            }
+          }
+        },
```
[the remark this answers is not in the document]
Comment (msywulak) [reply apps/docs/openapi.json:14173]: Fixed in e167a6e0 at the source: added \`required: true\` to the route's \`request.body\` in \`createRoute\` and regenerated \`apps/docs/openapi.json\` (hand-editing the generated JSON would drift on the next regen).

File: apps/docs/openapi.json [hunk 6]
```diff
@@ -14135,6 +14135,205 @@
...
+                    },
+                    "requestId": {
+                      "type": "string"
+                    }
+                  },
+                  "required": [
+                    "error",
+                    "message"
+                  ]
+                }
```
[the remark this answers is not in the document]
Comment (msywulak) [reply apps/docs/openapi.json:14305]: Declining this one: the 500 here uses the shared \`ErrorSchema\`, which is also the schema for the 404 branch on this same route — and that 404 (\`not_found\`) intentionally does not carry \`requestId\`. Marking \`requestId\` required would either force a hand-edit to generated JSON (drifts on regen) or a bespoke per-status schema diverging from the shared one. The 500 handler does include \`requestId\` at runtime for log correlation.

File: packages/api/src/api/routes/wizard.ts [hunk 7]
```diff
@@ -676,6 +753,123 @@ wizard.openapi(generateRoute, async (c) => {
...
+      catch: (err) => (err instanceof Error ? err : new Error(String(err))),
+    }).pipe(Effect.either);
+
+    if (enrichResult._tag === "Left") {
+      const err = enrichResult.left;
+      log.error({ err, requestId, connectionId, tableName }, "Wizard enrich failed");
+      return c.json({
+        error: "enrich_failed",
+        message: `Failed to enrich "${tableName}": ${err.message}`,
+        requestId,
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/api/routes/wizard.ts:858]: Fixed in e167a6e0. The 500 now returns a generic, actionable message + \`requestId\`; the full error (driver/provider detail) stays in \`log.error\`.

File: packages/api/src/lib/providers.ts [hunk 8]
```diff
@@ -186,6 +186,25 @@ export function getMissingProviderConfig(provider: string): string[] {
...
+ * {@link getModel} actually does at call time.
+ *
+ * Used by the wizard's two-phase generate enrich endpoint (issue #3236) to
+ * fail fast with one actionable "configure a provider" message instead of
+ * letting every per-table enrichment hit the same provider-auth error.
+ */
+export function getMissingModelConfig(): { provider: string; missing: string[] } {
+  const provider = process.env.ATLAS_PROVIDER ?? getDefaultProvider();
+  return { provider, missing: getMissingProviderConfig(provider) };
+}
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/api/src/lib/providers.ts:213]: Fixed in e167a6e0. \`getMissingModelConfig\` now returns a non-empty \`missing\` for an unsupported \`ATLAS_PROVIDER\` (via \`isSupportedProvider\`), so the preflight fails fast instead of reporting healthy and throwing at \`getModel()\` time. Added unit tests (unsupported / keyless / configured).

File: packages/web/src/app/wizard/page.tsx [hunk 9]
```diff
@@ -638,9 +690,149 @@ function StepReview({
...
+      },
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/web/src/app/wizard/page.tsx:869]: Fixed in e167a6e0 — \`applyEnrichedYaml\` applies the result only when the row still equals the YAML snapshot we sent, so a mid-flight manual edit is preserved. Also locked the textarea while the row is enriching as a belt-and-suspenders.

File: packages/web/src/app/wizard/page.tsx [hunk 10]
```diff
@@ -638,9 +690,149 @@ function StepReview({
...
+        if (aborted.current) return { kind: "skipped" };
+        const outcome = await enrichOne(tableName, yamlByTable.get(tableName) ?? "");
+        if (outcome.kind === "error" && outcome.unavailable) aborted.current = true;
+        return outcome;
+      },
+      (tableName, outcome) => {
+        if (!outcome || outcome.kind === "skipped") {
+          setEnrichStatus((prev) => ({ ...prev, [tableName]: "idle" }));
+          return;
+        }
```
[the remark this answers is not in the document]
Comment (msywulak) [reply packages/web/src/app/wizard/page.tsx]: Fixed in e167a6e0. \`onSettled\` now handles the third \`workerError\` arg explicitly — logs it and marks the row \`error\` with a message — instead of silently treating an unexpected throw as a skip.

**Comment by msywulak** [comment]
@codex review

**Comment by msywulak** [comment]
@coderabbitai review
